shap/utils/_clustering.py

```diff
@@ -169,7 +171,13 @@ def xgboost_distances_r2(
     return dist
 
 
-def hclust(X, y=None, linkage="single", metric="auto", random_state=0):
+def hclust(
+    X: _ArrayLike,
+    y: Optional[_ArrayLike] = None,
+    linkage: Literal["single", "complete", "average"] = "single",
+    metric: str = "auto",
+    random_state: int = 0,
```

Comment:
not a dealbreaker, but shall we also include the `RandomState` as part of the allowed types?

(Might even be worth doing a TypeAlias, since this pops up quite often in multiple places)

Reply:
Good shout, done. I decided against a type alias as it's a pretty simple type, and I think writing it in-line has better [Locality of Behaviour](https://htmx.org/essays/locality-of-behaviour/)